### PR TITLE
Hearings: refactor code

### PR DESCRIPTION
**Pull request by nickhill111**
SSCS-12058

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java [hunk 1]
```diff
@@ -451,19 +419,42 @@ public static List<UnavailabilityDayOfWeek> getDwpUnavailabilityDayOfWeek() {
...
+    private static UnavailabilityRange getUnavailabilityRange(DateRange dateRange) throws ListingException {
+        LocalDate startDate = dateRange.getStartDate();
+
+        if (isNull(startDate)) {
+            return null;
+        }
+
+        LocalDate endDate = dateRange.getEndDate();
+
+        if (endDate.isBefore(startDate)) {
```
Comment (DGordonV1) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java:449]: Will this return null or throw exception if end date is null or not valid date format?

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java [hunk 2]
```diff
@@ -1157,6 +1158,19 @@ void buildHearingPartiesDetails_unavailabilityRanges() throws InvalidMappingExce
         assertThat(unavailabilityRange.getUnavailableToDate()).isEqualTo(end);
     }
 
+    @DisplayName("When the start and end date ranges are both null, then return null")
```
Comment (DGordonV1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java:1161]: Is it worth covering when end date is null but start is fine or an invalid format on either dates as well?

File: src/main/java/uk/gov/hmcts/reform/sscs/service/ServiceHearingsService.java [hunk 3]
```diff
@@ -63,6 +63,15 @@ public ServiceHearingValues getServiceHearingValues(ServiceHearingRequest reques
         return model;
     }
 
+    private ServiceHearingValues getModel(SscsCaseData caseData) throws ListingException, UpdateCaseException {
```
Comment (DGordonV1) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/ServiceHearingsService.java:66]: Could we make this method name a bit less generic, getServiceHearingValuesModel or something like that?

**Review by DGordonV1 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/ServiceHearingsService.java (hunk 3 above)
Comment (nickhill111) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/ServiceHearingsService.java:66]: done

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java (hunk 2 above)
Comment (nickhill111) [reply src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java:1161]: added 2 extra tests for this as well as a refactor for commons

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java (hunk 1 above)
Comment (DGordonV1) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java:449]: Cool, so we're checking against this outcome via common changes. Good stuff.

**Review by DGordonV1 — DISMISSED** [review]

**Review by AndreiTrushkov — APPROVED** [review]

**Review by AndreiTrushkov — DISMISSED** [review]

**Review by DGordonV1 — APPROVED** [review]

**Review by AndreiTrushkov — APPROVED** [review]

**Pull request merged**